### PR TITLE
docs: add document for usage of pymeilisearch actions

### LINKED ISSUE #336: Document the usage of PyMeilisearch in CI/CD
💡 **Idea**
With [pymeilisearch](https://github.com/ansys/pymeilisearch) public, it is possible now for projects to upload indexed documentation.

Once https://github.com/ansys/actions/issues/294 is implemented, its usage with `conf.py` file should be indicated in our development guide.

### PR DESCRIPTION
**Pull request by Revathyvenugopal162**
Resolves #336

### CODE REVIEW

File: doc/source/how-to/code/docs.yml [hunk 1]
```diff
@@ -26,7 +26,21 @@ doc-deploy-stable:
...
+doc-deploy-index:
+  name: "Index the documentation and scrap using pymeilisearch"
+  runs-on: ubuntu-latest
+  needs: doc-deploy
+  if: github.event_name == 'push'
+  steps:
+    - name: "Deploy the latest documentation index"
+      uses: ansys/actions/doc-deploy-index@v4.1
+      with:
+          cname: "<library>.docs.pyansys.com/version/<version-number>"
```
Comment (RobPasMue) [inline doc/source/how-to/code/docs.yml:43]: Handling yet another location for the version number might be "undesired"... Is there any way we can automate this? I'd avoid having it in more than one location that version number...

File: doc/source/how-to/documenting.rst [hunk 2]
```diff
@@ -693,9 +693,73 @@ As you are making changes in this branch, you want to periodically generate the
...
+        needs: doc-deploy
+        if: github.event_name == 'push'
+        steps:
+          - name: Scrape the stable documentation to PyMeilisearch
+            run: |
+              VERSION=$(python -c "from <your-package> import __version__; print('.'.join(__version__.split('.')[:2]))")
+              VERSION_MEILI=$(python -c "from <your-package> import __version__; print('-'.join(__version__.split('.')[:2]))")
+              echo "Calculated VERSION: $VERSION"
+              echo "Calculated VERSION_MEILI: $VERSION_MEILI"
+
```
Comment (RobPasMue) [inline doc/source/how-to/documenting.rst:748]: See... we are adding it here in an intelligent way. Do the same in the previous workflow.

File: doc/source/how-to/releasing.rst [hunk 3]
```diff
@@ -427,7 +427,7 @@ the :ref:`Project approval and public release` section.
           runs-on: ubuntu-latest
           if: github.event_name == 'push' && contains(github.ref, 'refs/tags')
           steps:
-            - uses: pyansys/actions/release-github@v3
+            - uses: ansys/actions/release-github@v3
```
Comment (RobPasMue) [inline doc/source/how-to/releasing.rst:430]:
```suggestion
            - uses: ansys/actions/release-github@v4
```

There are some other "v3"s missing. Let me try to suggest them all

File: doc/source/how-to/releasing.rst [hunk 4]
```diff
@@ -396,7 +396,7 @@ Replace ``<PACKAGE-NAME>`` and ``<TOKEN-REDACTED>`` with the package name and th
           runs-on: ubuntu-latest
           if: github.event_name == 'push' && contains(github.ref, 'refs/tags')
           steps:
-            - uses: pyansys/actions/release-pypi-public@v3
+            - uses: ansys/actions/release-pypi-public@v3
```
Comment (RobPasMue) [inline doc/source/how-to/releasing.rst:399]:
```suggestion
            - uses: ansys/actions/release-pypi-public@v4
```

**Review by RobPasMue — CHANGES_REQUESTED** [review]

File: doc/source/how-to/documenting.rst (hunk 2 above)
Comment (Revathyvenugopal162) [reply doc/source/how-to/documenting.rst:748]: done thanks for pointing @RobPasMue

File: doc/source/how-to/code/docs.yml [hunk 5]
```diff
@@ -26,7 +26,29 @@ doc-deploy-stable:
...
+      uses: ansys/actions/doc-deploy-stable@v4
       with:
         cname: ${{ env.CNAME }}
         token: ${{ secrets.GITHUB_TOKEN }}
+
+jobs:
+  doc-deploy-index:
+    name: "Index the documentation and scrap using PyMeilisearch"
+    runs-on: ubuntu-latest
+    needs: doc-deploy
```
Comment (RobPasMue) [inline doc/source/how-to/code/docs.yml:38]:
```suggestion
  doc-deploy-index:
    name: "Index the documentation and scrap using PyMeilisearch"
    runs-on: ubuntu-latest
    needs: doc-deploy-dev
```

**Review by RobPasMue — APPROVED** [review]

**Comment by RobPasMue** [comment]
Feel free to merge whenever you want @Revathyvenugopal162

**Pull request merged**